### ISSUE TITLE
ATF/PTF invalid combination error

### ISSUE THREAD

**Issue by nkolev92** [NuGet/Home#5622]
We need to reenable this fix. 
https://github.com/NuGet/NuGet.Client/pull/1495
which was later reverted here:
https://github.com/NuGet/NuGet.Client/pull/1507

Original intent of the fix.
```
Projects with PackageTargetFallback and AssetTargetFallback will now fail to restore from Visual Studio with NU1003.

Updating PackageSpec to take both PTF and ATF instead of a bool, this allows restore to fail across all restore types, including Visual Studio in the same way.
NU1003 will now show up as an error that fails restore, but restore will still proceed using the default project framework without fallbacks. This allows system packages to be restored so that the error is not lost with missing reference errors.
Updating error message string with the new text from @anangaur
```

The reason why this fix was removed at first is because we couldn't insert into the CLI/SDK, there were inconsistencies in the reading of of the assets file which caused the first restore to fail. 

https://github.com/NuGet/Home/issues/5449

**Comment by emgarten** [comment]
Moving to 4.4, this will need a different fix since the file format will have shipped in 4.3.

**Comment by nkolev92** [comment]
Redundant now.
